src/hearings/containers/request-hearing/hearing-link/hearing-link.component.ts

```diff
@@ -0,0 +1,90 @@
+import { AfterViewInit, Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { select, Store } from '@ngrx/store';
+import { Observable, Subscription } from 'rxjs';
+import { HearingLinksStateData } from '../../../models/hearingLinksStateData.model';
+import { ACTION, HearingLinkMessages } from '../../../models/hearings.enum';
+import { ServiceLinkedCasesModel } from '../../../models/linkHearings.model';
+import { HearingsService } from '../../../services/hearings.service';
+import * as fromHearingStore from '../../../store';
+import { LoadServiceLinkedCases } from '../../../store';
+import { RequestHearingPageFlow } from '../request-hearing.page.flow';
+
+@Component({
+  selector: 'exui-hearing-link',
+  templateUrl: './hearing-link.component.html',
+  styleUrls: ['./hearing-link.component.scss']
+})
+export class HearingLinkComponent extends RequestHearingPageFlow implements OnInit, AfterViewInit, OnDestroy {
+  public caseId: string;
+  public hearingLinksStateData$: Observable<HearingLinksStateData>;
+  public linkedCases: ServiceLinkedCasesModel[];
+  public hearingLinkForm: FormGroup;
+  public validationErrors: { id: string, message: string }[] = [];
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>,
+              protected readonly hearingsService: HearingsService,
+              protected readonly route: ActivatedRoute,
+              private readonly formBuilder: FormBuilder) {
+    super(hearingStore, hearingsService);
+    this.hearingStore.pipe(select(fromHearingStore.getHearingsFeatureState)).subscribe(
+      state => {
+        this.caseId = state.hearingList.hearingListMainModel ? state.hearingList.hearingListMainModel.caseRef : '';
+      }
+    );
+  }
+
+  public ngOnInit(): void {
+    this.hearingLinkForm = this.formBuilder.group({
+      hearingLink: ['', Validators.required]
```

Comment:
by default it should also prepopulate what is in the store hearingDetails.hearingIsLinkedFlag

Reply:
Thank you @rayliangatos I have done the changes.